/*
main diff is the handling of the base_product
We replace the BaseProductRequest and BaseProductResponse to use the Oneof
instead of the mother class of the inheritance.

Steps:
1. replace the BaseProductRequest and BaseProductResponse from the main file by the one above
2. generate the pb2 file with this change from iot_controller_backend/srcs:
    python3 -m grpc_tools.protoc  -I=.  features/products_controller/grpc/products_controller.proto  --python_out=. --grpc_python_out=. --experimental_allow_proto3_optional
*/

// Polymorphism
// Manual
message BaseProductRequest {
  oneof product {
    CoffeeMachineRequest coffee_machine = 2;
    LedPanelRequest led_panel = 3;
  }
}

message BaseProductResponse {
  oneof product {
    CoffeeMachineResponse coffee_machine = 2;
    LedPanelResponse led_panel = 3;
  }
}
